app/screens/gallery/renderers/video/video_controls/progress_bar.tsx

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useRef} from 'react';
+import {StyleSheet, View} from 'react-native';
+import {Gesture, GestureDetector} from 'react-native-gesture-handler';
+import Animated, {
+    useAnimatedStyle,
+    useSharedValue,
+    runOnJS,
+    withTiming,
+    interpolate,
+    runOnUI,
+} from 'react-native-reanimated';
+
+interface ProgressBarProps {
+    progress: number;
+    duration: number;
+    onSeek: (time: number) => void;
+}
+
+const styles = StyleSheet.create({
+    container: {
+        top: 4,
+        marginBottom: 8,
+        paddingHorizontal: 4,
+        width: '100%',
+        flex: 1,
+    },
+    touchArea: {
+        height: 24,
+        justifyContent: 'center',
+        position: 'relative',
+    },
+    track: {
+        backgroundColor: 'rgba(255, 255, 255, 0.3)',
+        position: 'absolute',
+        left: 0,
+        right: 0,
+    },
+    bar: {
+        backgroundColor: 'white',
+        position: 'absolute',
+        left: 0,
+    },
+    thumb: {
+        position: 'absolute',
+        backgroundColor: 'white',
+        top: '50%',
+        marginTop: -8,
+        shadowColor: '#000',
+        shadowOffset: {width: 0, height: 2},
+        shadowOpacity: 0.3,
+        shadowRadius: 4,
+        elevation: 4,
+    },
+});
+
+const ProgressBar: React.FC<ProgressBarProps> = ({
+    progress,
+    duration,
+    onSeek,
+}) => {
+    const progressRef = useRef<View>(null);
+    const progressWidth = useSharedValue(0);
+    const dragPosition = useSharedValue(0);
+    const trackHeight = useSharedValue(4);
+    const thumbOpacity = useSharedValue(0);
+    const isDraggingShared = useSharedValue(false);
+    const localProgress = useSharedValue(progress);
+
+    React.useEffect(() => {
+        runOnUI(() => {
+            'worklet';
+            if (!isDraggingShared.value) {
+                localProgress.value = progress;
+            }
+        })();
+
+    // no need to add shared values to the dependency array
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [progress]);
```

Comment:
Also learning as I go here, but is this the same as using an animated reaction? Is there a benefit to using useAnimatedReaction instead?